### PR TITLE
Custom photon fields

### PR DESCRIPTION
**Pull request by lukasmerten**
This PR is based on @JulienDoerner's work  (#16) to streamline the production of tabulated files.

Changes of this PR will be needed to make custom photon fields *easily* available in CRPropa, by reducing the complexity of that task significantly.

**Main changes:**
- It comes with a calc_all.py module that allows to produce the default CRPropa-tar with a single python call.
- Removed calc_photonFields.py:
calc_photonFields.py had function definitions for all default photon fields. That differed slightly from definition in photonFields.py, where each photon field model has it's class definition. Functionality of the function has been moved to the classes.
- Furthermore, it adds automatic absolute path detection to all individual calc_* modules. This allows to consistently import them from different directories. This helped to clean up the upcoming custom photon field example.

**Attention**
Before merging---consequently updating the tar ball, which is downloaded by CRPropa---this PR should be double checked carefully, as it touches more or less all precalculated files. 
CRPropa automated tests are passing. 
However, some additional sanity checks would be probably good.

### CODE REVIEW

File: calc_photopionproduction.py [hunk 1]
```diff
@@ -91,3 +84,23 @@
...
     np.savetxt(fname, data, fmt=fmt, header=header)
+
+if __name__ == "__main__":
+    fields = [
+        photonField.CMB(),
+        photonField.EBL_Kneiske04(),
+        photonField.EBL_Stecker05(),
+        photonField.EBL_Franceschini08(),
+        photonField.EBL_Finke10(),
+        photonField.EBL_Dominguez11(),
```
Comment (rafaelab) [inline calc_photopionproduction.py:95]: I suggest we bring back the tables for the Domínguez upper/lower limit EBL. They were present a few years ago and then got lost for some reason.

**Review by rafaelab — COMMENTED** [review]
Hi @lukasmerten 
It looks good to me.
I suggest to add the calculation of tables for the Domínguez lower/upper EBL models.
Otherwise, the PR can be merged.

File: calc_photopionproduction.py (hunk 1 above)
Comment (lukasmerten) [reply calc_photopionproduction.py:95]: I added the upper and lower bounds for the default CRPropa data tar ball. However, I did not add them to the all the calc_* files.

**Comment by JulienDoerner** [comment]
I also added a check if the plot folder exists.

File: photonField.py [hunk 2]
```diff
@@ -1,34 +1,82 @@
...
-T_CMB = 2.72548  # CMB temperature [K]
+class PhotonField(object):
+    """Base class for photon fields"""
 
+    def __init__(self):
+        self.name = 'PhotonField'
+        self.info = 'Base class photon field'
+        self.energy = []
+        self.redshift = None
+        self.photonDensity = []
```
[the remark this answers is not in the document]
Comment (lukasmerten) [reply photonField.py]: Done

File: calc_pairproduction.py [hunk 3]
```diff
@@ -10,16 +10,11 @@
...
-Mpc = 3.08567758e22  # [m]
-c0 = 299792458.  # [m/s]
-r0 = 2.817940e-15  # classical electron radius [m]
-alpha = 7.297352e-3  # fine-structure constant
-me = 9.10938291e-31  # electron mass [kg]
-me_c2 = me * c0**2  # electron mass in [J/c^2]
-mp = 1.67262178e-27  # proton mass [kg]
+cdir = os.path.split(__file__)[0]
 
+me_c2 = mass_electron * c_squared  # electron mass in [J/c^2]
```
[the remark this answers is not in the document]
Comment (lukasmerten) [reply calc_pairproduction.py]: Done

File: calc_mass.py [hunk 4]
```diff
@@ -1,6 +1,9 @@
 from pylab import *
 import crpropa
```
[the remark this answers is not in the document]
Comment (lukasmerten) [reply calc_mass.py]: Done. I missed that when cleaning up the files.

File: calc_all.py [hunk 5]
```diff
@@ -0,0 +1,258 @@
...
+        f.write(checksum)
+    t2 = time.time()
+    print("\nChecksum calculated in {} seconds.".format(round(t2-t1, 2)))
+    print("#"*50+"\n")
+
+def createPhotonTargetInteractions(fields: list):
+    """Create all interaction tables that depend on a photon field
+    
+    Input
+        fields: list of list of CRPropa photonField class instances
```
[the remark this answers is not in the document]
Comment (lukasmerten) [reply calc_all.py]: Done

File: calc_all.py [hunk 6]
```diff
@@ -0,0 +1,258 @@
...
+        bh.process(field)
+    t2 = time.time()
+    print("\nBH pair production tables generated in {} seconds.".format(round(t2-t1, 2)))
+    print("#"*50+"\n")
+
+def photo_disintegration(rateFields: list, emissionFields: list):
+    """Calculate photo disintegration
+    
+    Input
+        fields: list of CRPropa photonField class instances
```
[the remark this answers is not in the document]
Comment (lukasmerten) [reply calc_all.py]: Done

File: calc_photopionproduction.py (hunk 1 above)
Comment (lukasmerten) [reply calc_photopionproduction.py]: Photon field models are now imported in calc_* files. So there is only one location that needs to be updated in the future.

**Comment by JulienDoerner** [comment]
For me, the PR looks good now. 
With merging this one we can also close the PR #7 as now the custom photon fields are fully implemented. 

But with adding the Dominguez upper and lower boundary to the data, we have to adjust the module in the CRPropa repository.

**Pull request merged**